conmon/conmon.c

```diff
@@ -878,9 +880,17 @@ static int setup_terminal_control_fifo()
 static void setup_oom_handling(int container_pid)
 {
 	/* Setup OOM notification for container process */
-	_cleanup_free_ char *memory_cgroup_path = process_cgroup_subsystem_path(container_pid, "memory");
+	_cleanup_free_ char *memory_cgroup_path = NULL;
 	_cleanup_close_ int cfd = -1;
+	struct statfs sfs;
 	int ofd = -1; /* Not closed */
+
+        if (statfs ("/sys/fs/cgroup", &sfs) == 0 && sfs.f_type == CGROUP2_SUPER_MAGIC) {
```

Comment:
will the statfs call complain if conmon is called as non-root?

Reply:
no, it works well.  I've tested it locally only as non root :-)